src/components/CategoriesPills.js

```diff
@@ -0,0 +1,144 @@
+// @flow
+import React from "react";
+import { View, StyleSheet, ScrollView } from "react-native";
+import LinearGradient from "react-native-linear-gradient";
+import type { StyleObj } from "react-native/Libraries/StyleSheet/StyleSheetTypes";
+import Text from "./Text";
+import text from "../constants/text";
+import eventCategories from "../constants/event-categories";
+import locale from "../data/locale";
+
+import {
+  blackColor,
+  whiteColor,
+  coralColor,
+  darkBlueGreyTwoColor,
+  eucalyptusGreenColor
+} from "../constants/colors";
+
+const categoryStyleColors = (category: string) => {
+  const categoryData = eventCategories[locale][category];
+
+  return {
+    backgroundColor: categoryData.color,
+    color: categoryData.contrast ? blackColor : whiteColor
+  };
+};
+
+type Props = {
+  selectedCategories: Set<string>,
+  style?: StyleObj
+};
+
+const CategoryPill = ({ name }: { name: string }) => (
+  <Text type="h3" style={[styles.categoryPill, categoryStyleColors(name)]}>
+    {name}
+  </Text>
+);
+
+class CategoriesPills extends React.PureComponent<Props> {
+  static defaultProps = {
+    style: {}
+  };
+
+  scrollView: ?Object;
+
+  scrollToLastPill = () => {
+    if (!this.scrollView) {
+      return;
+    }
+
+    this.scrollView.scrollToEnd({ animated: false });
+  };
+
+  render() {
+    const { style, selectedCategories } = this.props;
+
+    return (
+      <View style={[styles.selectedCategoriesPills, style]}>
+        {selectedCategories.size === 0 ? (
+          <Text type="h3" style={styles.zeroSelected}>
+            {text.zeroSelected}
+          </Text>
+        ) : (
+          <View style={styles.scrollView}>
+            <LinearGradient
+              style={styles.scrollShadowLeft}
+              end={{ x: 0, y: 0.5 }}
+              start={{ x: 1.0, y: 0.5 }}
+              colors={["rgba(0, 0, 0, 0)", "rgba(0, 0, 0, 0.3)"]}
+            />
+            <ScrollView
+              ref={(ref: ?Object) => {
+                this.scrollView = ref;
+              }}
+              horizontal
+              showsHorizontalScrollIndicator={false}
+              onContentSizeChange={this.scrollToLastPill}
+            >
+              {Array.from(selectedCategories).map(name => (
+                <CategoryPill key={name} name={name} />
+              ))}
+            </ScrollView>
+            <LinearGradient
+              style={styles.scrollShadowRight}
+              end={{ x: 1.0, y: 0.5 }}
+              start={{ x: 0, y: 0.5 }}
+              colors={["rgba(0, 0, 0, 0)", "rgba(0, 0, 0, 0.3)"]}
+            />
+          </View>
+        )}
+      </View>
+    );
+  }
+}
+
+const styles = StyleSheet.create({
+  selectedCategoriesPills: {
+    backgroundColor: darkBlueGreyTwoColor,
+    flex: 1,
+    height: 40,
```

Comment:
This should be `minHeight` instead. We need to allow the user to scale the font size up to 200%. If doing so, the pills need to get higher, otherwise the text is cropped.

For the shadow we need to find a way to use 100% of the height then, instead of fixed 40.